config/yaml.go

```diff
@@ -314,14 +334,9 @@ func unmarshalYAMLValue(reader io.ReadCloser, value interface{}, lookUp lookUpFu
 // will be used
 //
 // TODO: what if someone wanted a literal ${FOO} in config? need a small escape hatch
```

Comment:
At least after this PR, that syntax is only applicable in the values, right? I think most of the time "unwanted" expands of it are going to occur in comments for people explaining exactly what the `${FOO:BAR}` syntax means